lib/web/scripts/install.go

```diff
@@ -0,0 +1,158 @@
+/*
+ * Teleport
+ * Copyright (C) 2025  Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package scripts
+
+import (
+	"context"
+	_ "embed"
+	"strconv"
+	"strings"
+
+	"github.com/google/safetext/shsprintf"
+	"github.com/gravitational/trace"
+
+	"github.com/gravitational/teleport/lib/web/scripts/oneoff"
+)
+
+// AutoupdateStyle represents the kind of autoupdate mechanism the script should use.
+type AutoupdateStyle int
+
+const (
+	// NoAutoupdate means the installed Teleport should not autoupdate.
+	NoAutoupdate AutoupdateStyle = iota
+	// PackageManagerAutoupdate means the installed Teleport should update via a script triggering package manager
+	// updates. The script lives in the 'teleport-ent-update' package and was our original attempt at automatic updates.
+	PackageManagerAutoupdate
+	// UpdaterBinaryAutoupdate means the installed Teleport should update via the teleport-update binary.
+	// This update style does not depend on any package manager (although it has a system dependency to wake up the
+	// updater).
+	UpdaterBinaryAutoupdate
+)
+
+// InstallScriptOptions contains the Teleport installation options used to generate installation scripts.
+type InstallScriptOptions struct {
+	AutoupdateStyle AutoupdateStyle
+	// TeleportVersion that should be installed. Without the leading "v".
+	TeleportVersion string
+	// CDNBaseURL is the URL of the CDN hosting teleport tarballs.
+	// If left empty, the 'teleport-update' installer will pick the one to use.
+	CDNBaseURL string
```

Comment:
Can we please add an example or be more explicit about the expected format?
Eg, should it include the protocol? Should it include a leading slash?